src/app/qgslabelpropertydialog.cpp

```diff
@@ -344,6 +349,9 @@ void QgsLabelPropertyDialog::setDataDefinedValues( QgsVectorLayer *vlayer )
         }
         break;
       }
+      case QgsPalLayerSettings::MultiLineAlignment:
+        mMultiLineAlignComboBox->setCurrentIndex( mMultiLineAlignComboBox->findData( result.toString() ) );
```

Comment:
This should probably be case insensitive - case is ignored when evaluating the property